frontend/src/core/helpers.js

```diff
@@ -1,413 +1,336 @@
-          textArea.style.border = 'none';
-          textArea.style.outline = 'none';
-          textArea.style.boxShadow = 'none';
-    
-          // Avoid flash of white box if rendered for any reason.
-          textArea.style.background = 'transparent';
-    
-          document.body.appendChild(textArea);
-    
-          textArea.select();
-    
-          var success = document.execCommand('copy');
-
-          document.body.removeChild(textArea);
-          
-          return success;
-        },
-
-        // checks for at least one child object
-        validateCourseContent: function(currentCourse, callback) {
-          var containsAtLeastOneChild = true;
-          var alerts = [];
-          var iterateOverChildren = function(model, index, doneIterator) {
-            if(!model._childTypes) {
-              return doneIterator();
-            }
-            model.fetchChildren(function(currentChildren) {
-              if (currentChildren.length > 0) {
-                return helpers.forParallelAsync(currentChildren, iterateOverChildren, doneIterator);
-              }
-              containsAtLeastOneChild = false;
-              var children = _.isArray(model._childTypes) ? model._childTypes.join('/') : model._childTypes;
-              alerts.push(model.get('_type') + " '" + model.get('title') + "' missing " + children);
-              doneIterator();
-            });
-          };
-          // start recursion
-          iterateOverChildren(currentCourse, null, function() {
-            var errorMessage = "";
-            if(alerts.length > 0)  {
-              for(var i = 0, len = alerts.length; i < len; i++) {
-                errorMessage += "<li>" + alerts[i] + "</li>";
-              }
-              return callback(new Error(errorMessage));
-            }
-            callback(null, true);
-          });
-        },
-
-      isValidEmail: function(value) {
-        var regEx = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-        if (value.length === 0 || !regEx.test(value)) {
-          return false;
+      var k = 1000;
+      var sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+      var i = Math.floor(Math.log(bytes) / Math.log(k));
+
+      return (bytes / Math.pow(k, i)).toPrecision(3) + ' ' + sizes[i];
+    },
+
+    renderBooleanOptions: function(selectedValue) {
+      var options = ["true", "false"];
+      var html = '';
+
+      for (var i = 0; i < options.length; i++) {
+        var selected = selectedValue == options[i] ? ' selected' : '';
+        html += '<option value="' + options[i] + '"' + selected + '>' + options[i] + '</option>';
+      }
+      return new Handlebars.SafeString(html);
+    },
+
+    pickCSV: function (list, key, separator) {
+      var vals = [];
+      separator = (separator && separator.length) ? separator : ',';
+      if (list && list.length) {
```

Comment:
Looks like the logic hasn't been reversed here when returning early.

Anyhow, suggest refactoring to return right at the start of the function. Something like:
```js
pickCSV: function(list, key, separator) {
  if (!list || !list.length) {
    return '';
  }
  var vals = [];
  if (!separator || !separator.length) {
    separator = ',';
  }
  for (var i = 0, l = list.length; i < l; i++) {
    var val = list[i];
    var nestedVal = key && val[key];
    vals.push(nestedVal || val);
  }
  return vals.join(separator);
}
```